### ISSUE TITLE
How can i trust with output like this?

### ISSUE THREAD

**Issue by Jollym** [Microsoft/diskspd#15]
And this has popped up more then once already today.

Command Line: diskspd -d120 -F4 -w0 -r -b64k -o10 -h e:\temp\testfile.dat

Input parameters:

```
timespan:   1
-------------
duration: 120s
warm up time: 5s
cool down time: 0s
random seed: 0
path: 'e:\temp\testfile.dat'
    think time: 0ms
    burst size: 0
    software and hardware write cache disabled
    performing read test
    block size: 65536
    using random I/O (alignment: 65536)
    number of outstanding I/O operations: 10
    thread stride size: 0
    IO priority: normal
```

Results for timespan 1:

---

actual test time:   120.01s
thread count:       4
proc count:     32
## CPU |  Usage |  User  |  Kernel |  Idle

   0|   0.00%|   0.00%|    0.00%|  99.99%
   1|   0.00%|   0.00%|    0.00%|  99.99%
   2|   1.74%|   0.00%|    1.74%|  98.25%
   3|   0.00%|   0.00%|    0.00%|  99.99%
   4|   0.05%|   0.00%|    0.05%|  99.94%
   5|   0.00%|   0.00%|    0.00%|  99.99%
   6|   0.03%|   0.01%|    0.01%|  99.97%
   7|   1.54%|   0.73%|    0.81%|  98.46%
   8|   0.01%|   0.00%|    0.01%|  99.98%
   9|   0.00%|   0.00%|    0.00%|  99.99%
  10|   0.03%|   0.00%|    0.03%|  99.97%
  11|   0.00%|   0.00%|    0.00%|  99.99%
  12|   0.00%|   0.00%|    0.00%|  99.99%
  13|   0.00%|   0.00%|    0.00%|  99.99%
  14|   0.00%|   0.00%|    0.00%|  99.99%
  15|   0.00%|   0.00%|    0.00%|  99.99%
  16|   0.03%|   0.03%|    0.00%|  99.97%
  17|   0.00%|   0.00%|    0.00%|  99.99%
  18|   0.00%|   0.00%|    0.00%|  99.99%
  19|   0.09%|   0.00%|    0.09%|  99.90%
  20|   0.07%|   0.05%|    0.01%|  99.93%
  21|   2.64%|   0.69%|    1.95%|  97.35%
  22|   0.46%|   0.16%|    0.30%|  99.54%
  23|   0.00%|   0.00%|    0.00%|  99.99%
  24|   0.04%|   0.00%|    0.04%|  99.95%
  25|   0.01%|   0.00%|    0.01%|  99.98%
  26|   0.03%|   0.03%|    0.00%|  99.97%
  27|   0.00%|   0.00%|    0.00%|  99.99%
  28|   0.03%|   0.00%|    0.03%|  99.97%
  29|   0.00%|   0.00%|    0.00%|  99.99%
  30|   0.07%|   0.03%|    0.04%|  99.93%
##   31|   0.01%|   0.00%|    0.01%|  99.98%

avg.|   0.21%|   0.05%|    0.16%|  99.78%

Total IO
## thread |       bytes     |     I/Os     |     MB/s   |  I/O per s |  file

```
 0 |               0 |            0 |       0.00 |       0.00 | e:\temp\testfile.dat (50GB)
 1 |               0 |            0 |       0.00 |       0.00 | e:\temp\testfile.dat (50GB)
 2 |               0 |            0 |       0.00 |       0.00 | e:\temp\testfile.dat (50GB)
 3 |               0 |            0 |       0.00 |       0.00 | e:\temp\testfile.dat (50GB)
```

---

total:                 0 |            0 |       0.00 |       0.00

Read IO
## thread |       bytes     |     I/Os     |     MB/s   |  I/O per s |  file

```
 0 |               0 |            0 |       0.00 |       0.00 | e:\temp\testfile.dat (50GB)
 1 |               0 |            0 |       0.00 |       0.00 | e:\temp\testfile.dat (50GB)
 2 |               0 |            0 |       0.00 |       0.00 | e:\temp\testfile.dat (50GB)
 3 |               0 |            0 |       0.00 |       0.00 | e:\temp\testfile.dat (50GB)
```

---

total:                 0 |            0 |       0.00 |       0.00

Write IO
## thread |       bytes     |     I/Os     |     MB/s   |  I/O per s |  file

```
 0 |               0 |            0 |       0.00 |       0.00 | e:\temp\testfile.dat (50GB)
 1 |               0 |            0 |       0.00 |       0.00 | e:\temp\testfile.dat (50GB)
 2 |               0 |            0 |       0.00 |       0.00 | e:\temp\testfile.dat (50GB)
 3 |               0 |            0 |       0.00 |       0.00 | e:\temp\testfile.dat (50GB)
```

---

total:                 0 |            0 |       0.00 |       0.00

**Comment by Jollym** [comment]
More fun numbers

**Apparently there is a CPU when i look at C:\**


Command Line: diskspd -c50G -d120 -F4 -w0 -r -b64k -o10 c:\temp\testfile.dat

Input parameters:

	timespan:   1
	-------------
	duration: 120s
	warm up time: 5s
	cool down time: 0s
	random seed: 0
	path: 'c:\temp\testfile.dat'
		think time: 0ms
		burst size: 0
		using software and hardware write cache
		performing read test
		block size: 65536
		using random I/O (alignment: 65536)
		number of outstanding I/O operations: 10
		thread stride size: 0
		IO priority: normal



Results for timespan 1:
*******************************************************************************

actual test time:	120.00s
thread count:		4
proc count:		2

CPU |  Usage |  User  |  Kernel |  Idle
-------------------------------------------
   0|  21.95%|  12.69%|    9.26%|  78.04%
   1|  17.69%|  10.31%|    7.38%|  82.32%
-------------------------------------------
avg.|  19.82%|  11.50%|    8.32%|  80.18%

Total IO
thread |       bytes     |     I/Os     |     MB/s   |  I/O per s |  file
------------------------------------------------------------------------------
     0 |       631439360 |         9635 |       5.02 |      80.29 | c:\temp\testfile.dat (50GB)
     1 |       552665088 |         8433 |       4.39 |      70.27 | c:\temp\testfile.dat (50GB)
     2 |       627441664 |         9574 |       4.99 |      79.78 | c:\temp\testfile.dat (50GB)
     3 |       496893952 |         7582 |       3.95 |      63.18 | c:\temp\testfile.dat (50GB)
------------------------------------------------------------------------------
total:        2308440064 |        35224 |      18.35 |     293.53

Read IO
thread |       bytes     |     I/Os     |     MB/s   |  I/O per s |  file
------------------------------------------------------------------------------
     0 |       631439360 |         9635 |       5.02 |      80.29 | c:\temp\testfile.dat (50GB)
     1 |       552665088 |         8433 |       4.39 |      70.27 | c:\temp\testfile.dat (50GB)
     2 |       627441664 |         9574 |       4.99 |      79.78 | c:\temp\testfile.dat (50GB)
     3 |       496893952 |         7582 |       3.95 |      63.18 | c:\temp\testfile.dat (50GB)
------------------------------------------------------------------------------
total:        2308440064 |        35224 |      18.35 |     293.53

Write IO
thread |       bytes     |     I/Os     |     MB/s   |  I/O per s |  file
------------------------------------------------------------------------------
     0 |               0 |            0 |       0.00 |       0.00 | c:\temp\testfile.dat (50GB)
     1 |               0 |            0 |       0.00 |       0.00 | c:\temp\testfile.dat (50GB)
     2 |               0 |            0 |       0.00 |       0.00 | c:\temp\testfile.dat (50GB)
     3 |               0 |            0 |       0.00 |       0.00 | c:\temp\testfile.dat (50GB)
------------------------------------------------------------------------------
total:                 0 |            0 |       0.00 |       0.00


**But not when i look at D:\**

Command Line: diskspd -c50G -d120 -F4 -w0 -r -b64k -o10 d:\temp\testfile.dat

Input parameters:

	timespan:   1
	-------------
	duration: 120s
	warm up time: 5s
	cool down time: 0s
	random seed: 0
	path: 'd:\temp\testfile.dat'
		think time: 0ms
		burst size: 0
		using software and hardware write cache
		performing read test
		block size: 65536
		using random I/O (alignment: 65536)
		number of outstanding I/O operations: 10
		thread stride size: 0
		IO priority: normal



Results for timespan 1:
*******************************************************************************

actual test time:	195.25s
thread count:		4
proc count:		0

CPU |  Usage |  User  |  Kernel |  Idle
-------------------------------------------
-------------------------------------------
avg.|  -1.#J%|  -1.#J%|   -1.#J%|  -1.#J%

Total IO
thread |       bytes     |     I/Os     |     MB/s   |  I/O per s |  file
------------------------------------------------------------------------------
------------------------------------------------------------------------------
total:                 0 |            0 |       0.00 |       0.00

Read IO
thread |       bytes     |     I/Os     |     MB/s   |  I/O per s |  file
------------------------------------------------------------------------------
------------------------------------------------------------------------------
total:                 0 |            0 |       0.00 |       0.00

Write IO
thread |       bytes     |     I/Os     |     MB/s   |  I/O per s |  file
------------------------------------------------------------------------------
------------------------------------------------------------------------------
total:                 0 |            0 |       0.00 |       0.00



ETW:
----



          ETW Buffer Settings & Statistics
--------------------------------------------------------
(KB)                           Buffers   (Secs)   (Mins)
Size  | Min |  Max  |  Free  | Written  | Flush    Age
0         0     0      0          0        15        0

Allocated Buffers: 2
LOST EVENTS:      469762076
LOST LOG BUFFERS:     18932
LOST REAL TIME BUFFERS:16198024

**Comment by Jollym** [comment]
And since I'm on the subject that 100.01% doesn't look very Pro

CPU |  Usage |  User  |  Kernel |  Idle
-------------------------------------------
   0|   2.77%|   0.46%|    2.32%|  97.23%
   1|   3.42%|   0.38%|    3.05%|  96.58%
   2|   3.01%|   0.40%|    2.60%|  97.00%
   3|   2.99%|   0.31%|    2.68%|  97.01%
   4|   0.01%|   0.00%|    0.01%|  99.99%
   5|   0.00%|   0.00%|    0.00%| 100.01%
   6|   0.03%|   0.00%|    0.03%|  99.98%
   7|   1.41%|   0.55%|    0.86%|  98.60%
   8|   0.05%|   0.03%|    0.03%|  99.95%
   9|   0.00%|   0.00%|    0.00%| 100.01%
  10|   0.00%|   0.00%|    0.00%| 100.01%
  11|   0.00%|   0.00%|    0.00%| 100.01%
  12|   0.00%|   0.00%|    0.00%| 100.01%
  13|   0.00%|   0.00%|    0.00%| 100.01%
  14|   0.00%|   0.00%|    0.00%| 100.01%
  15|   0.00%|   0.00%|    0.00%| 100.01%
  16|   0.05%|   0.05%|    0.00%|  99.95%
  17|   0.00%|   0.00%|    0.00%| 100.01%
  18|   0.00%|   0.00%|    0.00%| 100.01%
  19|   0.01%|   0.00%|    0.01%|  99.99%
  20|   0.00%|   0.00%|    0.00%| 100.01%
  21|   2.46%|   0.56%|    1.90%|  97.54%
  22|   0.16%|   0.07%|    0.09%|  99.85%
  23|   0.00%|   0.00%|    0.00%| 100.01%
  24|   0.12%|   0.03%|    0.09%|  99.89%
  25|   0.00%|   0.00%|    0.00%| 100.01%
  26|   0.00%|   0.00%|    0.00%| 100.01%
  27|   0.00%|   0.00%|    0.00%| 100.01%
  28|   0.01%|   0.00%|    0.01%|  99.99%
  29|   0.00%|   0.00%|    0.00%| 100.01%
  30|   0.00%|   0.00%|    0.00%| 100.01%
  31|   0.00%|   0.00%|    0.00%| 100.01%
-------------------------------------------
avg.|   0.52%|   0.09%|    0.43%|  99.49%

**Comment by dl2n** [comment]
The 100.01% does look peculiar. I'm seeing some more variation now that I'm looking for it directly. The specific order of runtime capture v. performance counter capture might be one component of it, but I'm as suspicious of possible floating point error propagation in the math in the resultparser (shared with xmlresultparser) rendering. I'll poke at it.

Regarding the no-IO cases: what do you see at the PhysicalDisk performance counters for E:? Also zero?

**Comment by DanPear** [comment]
As the sum of the delta of user time and kernel time (which also includes idle time) accounts for all of the CPU time within the collection interval, its value should be used instead of the captured running time to avoid errors with conversions. The commit a59604ff104f1fe50a36cc1d4271491fc03430c8 offers a possible solution.

**Comment by DanPear** [comment]
This issue has been addressed in a48cc0d3613d23459f436a476b436a113146aeb7 and confirmed as working.
